File: 1984.学生分数的最小差值.js
/*
 * @lc app=leetcode.cn id=1984 lang=javascript
 *
 * [1984] 学生分数的最小差值
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function(nums, k) {
    if (k === 0) return 0;
    nums.sort((a, b) => a - b);
    let minDiff = Infinity;

    for (let i = 0; i <= nums.length - k; i++) {
        const currentDiff = nums[i + k - 1] - nums[i];
        minDiff = Math.min(minDiff, currentDiff);
    }

    return minDiff;
};
// @lc code=end

File: 1984.学生分数的最小差值.java
/*
 * @lc app=leetcode.cn id=1984 lang=java
 *
 * [1984] 学生分数的最小差值
 */

// @lc code=start
class Solution {
    public int minimumDifference(int[] nums, int k) {
        // 首先肯定要排序
        // 然后使用滑动窗口来找到最小差值
        Arrays.sort(nums);
        int n = nums.length;
        int minDiff = Integer.MAX_VALUE;
        for (int i = 0; i + k - 1 < n; i++) {
            // 计算当前窗口的差值
            int diff = nums[i + k - 1] - nums[i];
            // 更新最小差值
            minDiff = Math.min(minDiff, diff);
        }
        // 如果没有找到合适的窗口，返回 -1
        return minDiff == Integer.MAX_VALUE ? -1 : minDiff;
    }
}
// @lc code=end

